scripts/src/public/services/native_offers.ts

```diff
@@ -44,19 +44,42 @@ export function isPayToUser(jwt: ExternalOrderJWT): jwt is ExternalPayToUserOrde
 }
 
 export async function validateExternalOrderJWT(jwt: string, appUserId: string): Promise<ExternalOrderJWT> {
-	const decoded = await verifyJWT<PayToUserPayload, "spend" | "earn" | "pay_to_user">(jwt);
+	const decoded = await verifyJWT<Partial<PayToUserPayload>, "spend" | "earn" | "pay_to_user">(jwt);
 
 	if (decoded.payload.sub !== "earn" && decoded.payload.sub !== "spend" && decoded.payload.sub !== "pay_to_user") {
 		throw InvalidExternalOrderJwt();
 	}
 
-	if ((decoded.payload.sub === "spend" || decoded.payload.sub === "pay_to_user") &&
-		!!decoded.payload.sender.user_id && decoded.payload.sender.user_id !== appUserId) {
+	switch (decoded.payload.sub) {
+		case "spend":
+			if (!decoded.payload.sender) { throw MissingFieldJWT("sender"); }
+			if (!decoded.payload.offer) { throw MissingFieldJWT("offer"); }
+			break;
+
+		case "earn":
+			if (!decoded.payload.recipient) { throw MissingFieldJWT("recipient"); }
+			if (!decoded.payload.offer) { throw MissingFieldJWT("offer"); }
+			break;
+
+		case "pay_to_user":
+			if (!decoded.payload.sender) { throw MissingFieldJWT("sender"); }
+			if (!decoded.payload.recipient) { throw MissingFieldJWT("recipient"); }
+			if (!decoded.payload.offer) { throw MissingFieldJWT("offer"); }
+			break;
+
+		default: break;
+	}
+
+	if (
+		(decoded.payload.sub === "spend" || decoded.payload.sub === "pay_to_user")
+		&& decoded.payload.sender && !!decoded.payload.sender.user_id
+		&& decoded.payload.sender.user_id !== appUserId
+	) {
 		// if sender.user_id is defined and is different than current user, raise error
 		throw ExternalEarnOfferByDifferentUser(appUserId, decoded.payload.sender.user_id);
 	}
 
-	if (decoded.payload.sub === "earn" && decoded.payload.recipient.user_id !== appUserId) {
+	if (decoded.payload.sub === "earn" && decoded.payload.recipient && decoded.payload.recipient.user_id !== appUserId) {
```

Comment:
I think it should be:
```typescript
decoded.payload.sub === "earn" && (!decoded.payload.recipient || decoded.payload.recipient.user_id !== appUserId)
```

Reply:
@doodyparizada 
no, this is actually correct. he checks if `decoded.payload.recipient` is missing above (and if so throws a different error: `MissingFieldJWT("recipient");`

Reply:
So remove `&& decoded.payload.recipient ` check completely. It being here is wrong. If the check above is removed, this code will allow earn JWT that are missing the `recipient` key.

Reply:
Removed

Reply:
👍